### PR TITLE
Refactor: Modularize VPC and Security Group resources

### PR DESCRIPTION
**Pull request by spartan-longnguyen**
Reorganized the VPC and Security Group code to improve modularity and maintainability. Moved `security-group` resources into a dedicated module and updated related examples, documentation, and outputs accordingly. Adjusted paths and removed deprecated or redundant configurations to streamline resource management.

## Summary

- Reorganized the VPC and Security Group code to improve modularity and maintainability. 
- Moved `security-group` resources into a dedicated module and updated related examples, documentation, and outputs accordingly. 
- Adjusted paths and removed deprecated or redundant configurations to streamline resource management.

## Types of Changes

- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🕷 Bug fix (non-breaking change which fixes an issue)
- [ ] 👏 Performance optimization (non-breaking change which addresses a performance issue)
- [x] 🛠 Refactor (non-breaking change which does not change existing behavior or add new functionality)
- [ ] 📗 Library update (non-breaking change that will update one or more libraries to newer versions)
- [ ] 📝 Documentation (non-breaking change that doesn't change code behavior, can skip testing)
- [ ] ✅ Test (non-breaking change related to testing)
- [ ] 🔒 Security awareness (changes that effect permission scope, security scenarios)

## Test Plan
Terraform plan

### CODE REVIEW

File: aws/security-group/variables.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    egress_rules        = list(string)
+    egress_cidr_blocks  = list(string)
+  }))
+}
+
+variable "rules" {
+  description = "Map of known security group rules (define as 'name' = ['from port', 'to port', 'protocol', 'description'])"
+  type        = map(list(any))
+
+  default = {
```
Comment (spartan-toannhan) [inline aws/security-group/variables.tf:18]: default should be empty rules @spartan-longnguyen

File: aws/vpc/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to this project will be documented in this file.
 
+## [0.1.21]() (2024-12-24)
+### Features
+* Separate the VPC and Security Group
```
Comment (spartan-ytruong) [inline aws/vpc/CHANGELOG.md:7]: I would suggest a minor change here, to v0.2.0

File: aws/vpc/README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ You can specify `enable_single_nat=true` to enable only one NAT gateway for all
 ### Create a VPC
 ```hcl
 module "vpc" {
-  source  = "github.com/spartan-stratos/terraform-modules//aws/vpc?ref=v0.1.0"
+  source  = "github.com/spartan-stratos/terraform-modules//aws/network/vpc?ref=v0.1.0"
```
Comment (spartan-ytruong) [inline aws/vpc/README.md:19]: missing typo error

File: aws/security-group/examples/complete/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# VPC
```
Comment (spartan-ytruong) [inline aws/security-group/examples/complete/README.md:1]: missing typo error

File: aws/security-group/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+## [0.1.20]() (2024-12-24)
+### Features
+* Initialize Security Groups modules
```
Comment (spartan-ytruong) [inline aws/security-group/CHANGELOG.md:7]: Please also consider this as a minor change along with VPC module changes

File: aws/security-group/README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+This module will create the following components:
+- AWS Security Groups
+- Security Group Rules (Ingress and Egress)
+
+## Usage
+### Create Security Groups
+```hcl
+module "security_groups" {
+  source = "github.com/spartan-stratos/terraform-modules//aws/security-group?ref=v0.1.21"
```
Comment (spartan-ytruong) [inline aws/security-group/README.md:12]: please update to v0.1.22

File: aws/security-group/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      ingress_self             = [true]
+      egress_rules             = ["https"]
+      egress_cidr_blocks       = ["0.0.0.0/0"]
+      egress_ipv6_cidr_blocks  = []
+      egress_self              = []
+    }
+  ]
+
+  rules = {
+    "allow-all" = [0, 0, "-1", "example-rule"]
```
Comment (spartan-ytruong) [inline aws/security-group/README.md:31]: can you add more rules here as examples, and update example main.tf too, thanks

**Review by spartan-ytruong — APPROVED** [review]

**Pull request merged**